### PR TITLE
Fixed corruption of import GnuPG binary keys by the ansible.builtin.apt_key module, and related fixes

### LINKED ISSUE #74423: Invalid error message on GnuPG key import error in the apt_key module
### Summary

When importing a key by the `apt_key` module, in case of error, the error message was incorrect, because the logic to distinguish between the file name and the explicit data was inversed.

The error message was:

`
Unable to extract key from '-'
`

While it should have been:

`
Unable to extract key from 'inline data'
`

The relevant code is in the following line from: `/lib/ansible/modules/apt_key.py`

`
module.fail_json(msg="Unable to extract key from '%s'" % ('inline data' if data is None else filename), stdout=out, stderr=err)
`
(it should have been `if data is None`)

### Issue Type

Bug Report

### Component Name

ansible.builtin.apt_key module

### Ansible Version

```console
2.12.0.dev0 (latest revision from GitHub)
2.11.0rc2 (from pip install ansible)
```

### Configuration

```console
config not changed
```

### OS / Environment

Ubuntu 18

### Steps to Reproduce

To reproduce this error, create the following files:

The `Vagrantfile` file:

```
Vagrant.configure("2") do |config|
  config.vm.base_mac = nil
  config.vm.synced_folder ".", "/vagrant", disabled: false

  config.vm.define "test-host" do |n|
    n.vm.box = "ubuntu/bionic64"
    n.vm.hostname = "test-host"
    n.vm.provision :ansible do |ansible|
      ansible.limit = "all"
      ansible.playbook = "test.yaml"
    end
  end
end

```
The `test.yaml` file:

```
- name: Test Playbook
  hosts: test-host
  become: yes
  tasks:
    - name: Add binary key
      apt_key: url=https://packages.cloud.google.com/apt/doc/apt-key.gpg
```

and run `vagrant up`

### Expected Results

`
Unable to extract key from 'inline data'
`

### Actual Results

`
Unable to extract key from '-'
`

Relevant file: 
lib/ansible/modules/apt_key.py

@jvantuyl - please review
@Akasurde - please review

Fixed by https://github.com/ansible/ansible/pull/74476

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### LINKED ISSUE #74424: The ansible.builtin.apt_key module corrupts binary keys before importing them so import always fails
### Summary

When importing a key by the `apt_key `module, it gets corrupted and not imported in the case of a binary key (not a textual “armored” key). Ansible tries to convert string data, but some data is lost on conversion, which results in a GnuPG error. This is a bug because binary data should not be converted using the `to_native` function.

### Issue Type

Bug Report

### Component Name

ansible.builtin.apt_key module

### Ansible Version

```console
2.12.0.dev0 (latest revision from the GitHub repository) and 2.11.0rc2 (from pip install ansible)
```

### Configuration

```console
config not changed
```

### OS / Environment

Ubuntu 18

### Steps to Reproduce

To reproduce this error, create the following files:

Fie `Vagrantfile` file:

```
Vagrant.configure("2") do |config|
  config.vm.base_mac = nil
  config.vm.synced_folder ".", "/vagrant", disabled: false

  config.vm.define "test-host" do |n|
    n.vm.box = "ubuntu/bionic64"
    n.vm.hostname = "test-host"
    n.vm.provision :ansible do |ansible|
      ansible.limit = "all"
      ansible.playbook = "test.yaml"
    end
  end
end
```

The `test.yaml` file:

```
- name: Test Playbook
  hosts: test-host
  become: yes
  tasks:
    - name: Add binary key
      apt_key: url=https://packages.cloud.google.com/apt/doc/apt-key.gpg
      register: apt_key_binary_test

    - name: Validate the results
      assert:
        that:
          - 'apt_key_binary_test.changed is defined'
          - 'apt_key_binary_test.changed'
```

and run `vagrant up`

### Expected Results

Key imported without a problem

### Actual Results

```console
fatal: [master]: FAILED! => {"changed": false, "msg": "Unable to extract key from '-'", "stderr": "gpg: WARNING: no command supplied.  Trying to guess what you mean ...\ngpg: [don't know]: invalid packet (ctb=0a)\n", "stderr_lines": ["gpg: WARNING: no command supplied.  Trying to guess what you mean ...", "gpg: [don't know]: invalid packet (ctb=0a)"], "stdout": "pub:-:2048:1:FEEA9169307EA071:1614614617:1677728521::-:\nuid:::::::::Rapture Automatic Signing Key (cloud-rapture-signing-key-2021-03-01-08_01_09.pub):\nsub:-:2048:1:AA42F36EE8BEEE0E:1614614617::::\npub:-:2048:1:8B57C5C2836F4BEB:1607040606:1670154510::-:\nuid:::::::::gLinux Rapture Automatic Signing Key (//depot/google3/production/borg/cloud-rapture/keys/cloud-rapture-pubkeys/cloud-rapture-signing-key-2020-12-03-16_08_05.pub) <glinux-team@google.com>:\nsub:-:2048:1:48419E688DD52AC0:1607040606::::\n", "stdout_lines": ["pub:-:2048:1:FEEA9169307EA071:1614614617:1677728521::-:", "uid:::::::::Rapture Automatic Signing Key (cloud-rapture-signing-key-2021-03-01-08_01_09.pub):", "sub:-:2048:1:AA42F36EE8BEEE0E:1614614617::::", "pub:-:2048:1:8B57C5C2836F4BEB:1607040606:1670154510::-:", "uid:::::::::gLinux Rapture Automatic Signing Key (//depot/google3/production/borg/cloud-rapture/keys/cloud-rapture-pubkeys/cloud-rapture-signing-key-2020-12-03-16_08_05.pub) <glinux-team@google.com>:", "sub:-:2048:1:48419E688DD52AC0:1607040606::::"]}
```

Relevant file: 
lib/ansible/modules/apt_key.py

@jvantuyl - please review
@Akasurde - please review

Fixed by https://github.com/ansible/ansible/pull/74474

How did I encounter this error?

@fourstepper 

See https://packages.cloud.google.com/apt/doc/ 

The maintainers of this resource have changed the format of the `apt-key.gpg` file from text to binary on April 22nd, 2021. In the other words, they saved the text version of the key under the new name, `apt-key.gpg.asc`, and put the binary key under the old name. Meanwhile, the rest of the files with the .gpg extension remained text. For this reason, before April 22nd, playbooks that link to this URL directly worked and then stopped. You can replace the URL in the playbook with https://packages.cloud.google.com/apt/doc/apt-key.gpg.asc so that your .yaml works on current Ansible versions and not just on the latest pull from Github.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### LINKED ISSUE #74477: The 'ansible.builtin.apt_key' module does not always check for 'gpg' return codes
### Summary

Not all GnuPG return codes are analyzed (rc != 0) by the 'ansible.builtin.apt_key' module.

Relevant file:
lib/ansible/modules/apt_key.py

### Issue Type

Bug Report

### Component Name

ansible.builtin.apt_key

### Ansible Version

```console
2.12.0.dev0
2.11.0rc2
```

### Configuration

```console
Not changed
```

### OS / Environment

Any OS

### Steps to Reproduce

Fixed by: https://github.com/ansible/ansible/pull/74478

### Expected Results

I expected all calls of the 'gpg' binary have accompanying checks of the rc code. And if (rc != 0), then all relevant results are returned, including stdout and stderr, and also the command line with which the 'gpg' was called.

### Actual Results

The command line with which 'gpg' was called is not always returned as information, and the return code of 'gpg' is not always checked, and the stdout/stderr is not always returned as information.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by maximmasiutin**
1. Fixed corruption of import GnuPG binary keys. Fixes https://github.com/ansible/ansible/issues/74424 
2. The error message was incorrect because of a mixed logic bug. Fixes https://github.com/ansible/ansible/issues/74423 
3. The command line of GnuPG was logged in some cases but not logged in the other -- now it is logged in all cases. Fixes https://github.com/ansible/ansible/issues/74477
4. Some exceptions were not logged
5. Long strings made shorter to avoid pylint warnings
6. Fixed a pylint warning that the function did not always return a value
7. Not all GnuPG return codes were analyzed `(rc != 0)`  and logged. Fixes https://github.com/ansible/ansible/issues/74477
8. Minor pylint issues resolved like `(key_id_len != 8 and key_id_len != 16)` replaced to `(key_id_len not in (8, 16))`

### CODE REVIEW

**Comment by maximmasiutin** [comment]
ready_for_review

**Comment by Akasurde** [comment]
@maximmasiutin Thanks for the contribution. Could you please add a [changelog entry](https://github.com/ansible/ansible/tree/devel/changelogs/fragments) for this change and integration test [here](https://github.com/ansible/ansible/tree/devel/test/integration/targets/apt_key)? Thanks.

File: test/integration/targets/apt_key/tasks/main.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Test code for the apt_key module.
-# (c) 2017, James Tanner <tanner.jc@gmail.com>
+# (c) 2017, 2021 James Tanner <tanner.jc@gmail.com>
```
Comment (Akasurde) [inline test/integration/targets/apt_key/tasks/main.yml]:
```suggestion
# (c) 2017 James Tanner <tanner.jc@gmail.com>
```

We decided that we will only keep original year of copyright.

Comment (maximmasiutin) [reply test/integration/targets/apt_key/tasks/main.yml]: OK, thank you! I will fix that!

**Comment by maximmasiutin** [comment]
@Akasurde -- all checks have passed (127 successful checks) - could you please review again?

**Comment by Akasurde** [comment]
@maximmasiutin Could you please divide the changes into several PRs (depending upon bug fixing and formatting) so that it would be easier for code review and backporting the changes? Thanks.

**Comment by maximmasiutin** [comment]
@Akasurde 

I can make 3 pull requests:
1) Fixed corruption of import GnuPG binary keys. Fixes #74424
2) The error message was incorrect because of a mixed logic bug. Fixes #74423
3) The command line of GnuPG was logged in some cases but not logged in the other -- now it is logged in all cases; Some exceptions were not logged; Not all GnuPG return codes were analyzed (rc != 0) and logged

As about the formatting changes, I would not do that PR at all, since it is not that important.

However, the test case will be for the first pull request only, OK? Or should I make a separate test case for each of the pull requests, i.e., for the second and third pull requests as well?

Are you OK with that?

**Comment by Akasurde** [comment]
> @Akasurde
> 
> I can make 3 pull requests:
> 
>     1. Fixed corruption of import GnuPG binary keys. Fixes #74424
> 
>     2. The error message was incorrect because of a mixed logic bug. Fixes #74423
> 
>     3. The command line of GnuPG was logged in some cases but not logged in the other -- now it is logged in all cases; Some exceptions were not logged; Not all GnuPG return codes were analyzed (rc != 0) and logged
> 
> 
> As about the formatting changes, I would not do that PR at all, since it is not that important.
> 
> However, the test case will be for the first pull request only, OK? Or should I make a separate test case for each of the pull requests, i.e., for the second and third pull requests as well?
> 
> Are you OK with that?

Yes, that would be great.

**Comment by maximmasiutin** [comment]
@Akasurde - Thank you! I will do that. I will also try to write a test playbook for the second and the third issue from the above list - will just try to import a deliberately corrupted GnuPG binary key and check whether the error message is correct. For the second issue, it should return the string 'inline data', and for the third - the command line options for the GnuPG.

**Comment by maximmasiutin** [comment]
Will create 3 different pull requests.

**Comment by maximmasiutin** [comment]
First split: https://github.com/ansible/ansible/pull/74474

**Pull request closed without merging**